lib/csetgrp.gi

```diff
@@ -1119,39 +1193,49 @@ local c, flip, maxidx, refineChainActionLimit, cano, tryfct, p, r, t,
 	cnt:=indx;
       fi;
 
+      lstgens:=GeneratorsOfGroup(lst);
+      if Length(lstgens)>2 then
+        lstgens:=SmallGeneratingSet(lst);
+      fi;
+      lstgensop:=List(lstgens,i->i^pinv); # conjugate generators: operation
+      # is on cosets a.p; we keep original cosets: Ua.p.g/p, this
+      # corresponds to conjugate operation
+
+      if hom<>fail then
+        lstgensop:=List(lstgensop,i->Image(hom,i));
+      fi;
+
+      posi:=1;
       while bsz>0 and cnt>0 do
 	cnt:=cnt-1;
 
 	# compute orbit and stabilizers for the next step
         # own Orbitalgorithm and stabilizer computation
 
-	ps:=Position(blist,false);
+	#ps:=Position(blist,false);
+        while blist[posi] do posi:=posi+1;od;
+        ps:=posi;
```

Comment:
Or you could use `ps:=Position(blist,false,posi-1);` -- this is shorter and uses a kernel function (probably doesn't matter here, though).
Also, I think that if you change `posi:=1` to `posi:=0` above, you could also use `ps:=Position(blist,false,posi);`

Reply:
Thanks.